integration/kubernetes/helm-chart/alluxio/templates/master/statefulset.yaml

```diff
@@ -50,8 +50,8 @@ spec:
         heritage: {{ .Release.Service }}
         role: alluxio-master
     spec:
-      hostNetwork: {{ .Values.master.hostNetwork }}
-      dnsPolicy: {{ .Values.master.dnsPolicy }}
+      hostNetwork: {{ hasKey .Values.master "hostNetwork" | ternary .Values.master.hostNetwork $isRoot }}
```

Comment:
Use the specified `.Values.master.hostNetwork` if specified, otherwise infer based on whether the user is root.

FYI for boolean values the `{{ .Values.master.hostNetwork | default $isRoot}}` grammar doesn't work, because a false value for `.Values.master.hostNetwork` will also go to the default branch.